pkg/controller/releasemanager/state.go

```diff
@@ -296,6 +299,9 @@ func Canarying(ctx context.Context, deployment Deployment) (State, error) {
 	if deployment.isAlarmTriggered() {
 		return failing, nil
```

Comment:
you might want to delete on exit instead of in Canaried since there are multiple exits